sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -55,4 +62,32 @@ object JacksonUtils {
 
     schema.foreach(field => verifyType(field.name, field.dataType))
   }
+
+  private def validateStringLiteral(exp: Expression): String = exp match {
+    case Literal(s, StringType) => s.toString
+    case e => throw new AnalysisException("Must be a string literal, but: " + e)
+  }
+
+  def validateSchemaLiteral(exp: Expression): StructType =
+    DataType.fromJson(validateStringLiteral(exp)).asInstanceOf[StructType]
+
+  /**
+   * Convert a literal including a json option string (e.g., '{"mode": "PERMISSIVE", ...}')
+   * to Map-type data.
+   */
+  def validateOptionsLiteral(exp: Expression): Map[String, String] = {
+    val json = validateStringLiteral(exp)
+    parse(json) match {
+      case JObject(options) =>
```

Comment:
We could try to utilize `val parse(json).extract[Map[String, String]]`. Given my observation, it produces empty `Map` if there are no such values or empty json and it throws an exception if it is an invalid json.